### PR TITLE
fix: missing asset protocol path

### PR DESCRIPTION
**Pull request by wusyong**
**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix
- [ ] Feature
- [ ] Docs
- [ ] New Binding Issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes. Issue #___
- [x] No

**The PR fulfills these requirements:**

- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix: #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

### CODE REVIEW

**Review by lucasfernog — APPROVED** [review]

**Comment by wusyong** [comment]
Let me do some final check to see if it's not required on mac/linux

File: core/tauri/src/manager.rs [hunk 1]
```diff
@@ -284,6 +284,9 @@ impl<R: Runtime> WindowManager<R> {
     }
     if !webview_attributes.has_uri_scheme_protocol("asset") {
       webview_attributes = webview_attributes.register_uri_scheme_protocol("asset", move |url| {
+        #[cfg(target_os = "windows")]
+        let path = url.replace("asset://localhost/", "");
```
Comment (amrbashir) [inline core/tauri/src/manager.rs:288]: shouldn't it be 
```suggestion
        let path = url.replace("https://asset.localhost/", "");
```

Comment (wusyong) [reply core/tauri/src/manager.rs:288]: No this is in wry's context. Wry already handle that. The URL we get here will be `asset://...`

Comment (amrbashir) [reply core/tauri/src/manager.rs:288]: oh ok. didn't know that.

Comment (wusyong) [reply core/tauri/src/manager.rs:288]: But I suppose I should update the change file since it's windows only.

**Comment by ArturKovacs** [comment]
Is it true that the application code will have to produce `asset://localhost/path/to/file` on Windows but it must produce `asset://path/to/file` on every other system? If this is true what's the reasoning behind this distinction?

It seems simpler to just have to specify `asset://path/to/file` on every platform.

**Comment by lucasfernog** [comment]
Windows doesn't support custom protocols yet (such as `asset://`). That's why we have a `convertFileSrc` helper API.

**Comment by ArturKovacs** [comment]
Sorry I may have been confused about how this should be used. So if I understand it correctly my javascript should look something like this:
```js
let video = new HTMLVideoElement();
video.src = convertFileSrc("C:/path/to/my video.mp4");
```

**Comment by wusyong** [comment]
@ArturKovacs yes.
Btw we are looking into ways for streaming data. So you won't have to get entire binary for the video.

**Comment by ArturKovacs** [comment]
Sounds great, thank you!

**Review by lucasfernog — APPROVED** [review]

**Pull request merged**